packages/dev/core/src/Engines/Extensions/engine.multiRender.ts

```diff
@@ -108,44 +108,7 @@ ThinEngine.prototype.unBindMultiColorAttachmentFramebuffer = function (
 ): void {
     this._currentRenderTarget = null;
 
-    // If MSAA, we need to bitblt back to main texture
-    const gl = this._gl;
-
-    const attachments = rtWrapper._attachments!;
-    const count = attachments.length;
-
-    if (rtWrapper._MSAAFramebuffer) {
-        gl.bindFramebuffer(gl.READ_FRAMEBUFFER, rtWrapper._MSAAFramebuffer);
-        gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, rtWrapper._framebuffer);
-
-        for (let i = 0; i < count; i++) {
-            const texture = rtWrapper.textures![i];
-
-            for (let j = 0; j < count; j++) {
-                attachments[j] = gl.NONE;
-            }
-
-            attachments[i] = (<any>gl)[this.webGLVersion > 1 ? "COLOR_ATTACHMENT" + i : "COLOR_ATTACHMENT" + i + "_WEBGL"];
-            gl.readBuffer(attachments[i]);
-            gl.drawBuffers(attachments);
-            gl.blitFramebuffer(0, 0, texture.width, texture.height, 0, 0, texture.width, texture.height, gl.COLOR_BUFFER_BIT, gl.NEAREST);
-        }
-
-        for (let i = 0; i < count; i++) {
-            attachments[i] = (<any>gl)[this.webGLVersion > 1 ? "COLOR_ATTACHMENT" + i : "COLOR_ATTACHMENT" + i + "_WEBGL"];
-        }
-
-        gl.drawBuffers(attachments);
-    }
-
-    for (let i = 0; i < count; i++) {
-        const texture = rtWrapper.textures![i];
-        if (texture?.generateMipMaps && !disableGenerateMipMaps && !texture?.isCube && !texture?.is3D) {
-            this._bindTextureDirectly(gl.TEXTURE_2D, texture, true);
-            gl.generateMipmap(gl.TEXTURE_2D);
-            this._bindTextureDirectly(gl.TEXTURE_2D, null);
-        }
-    }
+    this.resolveFramebuffer(rtWrapper, disableGenerateMipMaps);

```

Comment:
Note that `unBindMultiColorAttachmentFramebuffer` is not needed anymore, but because it is public, I had to keep it.